custom_components/ha-google-home/api.py

```diff
@@ -110,16 +130,15 @@ async def get_alarms_and_timers(self, url, device):
                 )
         return resp
 
-    async def get_google_devices_information(self, zeroconf_instance):
-        """Retrieves devices from glocaltokens and fetches alarm/timer data from each of the device"""
+    async def update_google_devices_information(self):
+        """Retrieves devices from google home devices"""
 
         _LOGGER.debug("Fetching sensor data...")
 
         offline_devices = []
-        devices = await self.get_google_devices(zeroconf_instance)
         coordinator_data = {}
 
-        for device in devices:
+        for device in await self.get_google_devices():
```

Comment:
Processing awaitables in a loop is bad practice. Create async function to fetch alarms for single device and then use `asyncio.gather` to run it for all devices in parallel:
```python
devices = await self.get_google_devices(zeroconf_instance)
coordinator_data = await asyncio.gather([fetch_alarms(device) for device in devices]
```

Reply:
uuu, that's a really nice improvement, i was wondering how to shoot those tasks in parallel, thanks a lot! :boom: